File: app/src/main/java/io/github/wulkanowy/ui/splash/SplashPresenter.java
package io.github.wulkanowy.ui.splash;

import android.support.annotation.NonNull;

import javax.inject.Inject;

import io.github.wulkanowy.data.RepositoryContract;
import io.github.wulkanowy.ui.base.BasePresenter;

public class SplashPresenter extends BasePresenter<SplashContract.View>
        implements SplashContract.Presenter {

    @Inject
    SplashPresenter(RepositoryContract repository) {
        super(repository);
    }

    @Override
    public void attachView(@NonNull SplashContract.View view) {
        super.attachView(view);
        getView().setCurrentThemeMode(getRepository().getSharedRepo().getCurrentTheme());
        getView().cancelNotifications();

        if (getRepository().getSharedRepo().isUserLoggedIn()) {
            getView().openMainActivity();
        } else {
            getView().openLoginActivity();
        }
    }
}

File: app/src/main/java/io/github/wulkanowy/ui/splash/SplashPresenter.kt
package io.github.wulkanowy.ui.splash

import io.github.wulkanowy.data.ErrorHandler
import io.github.wulkanowy.data.repositories.SessionRepository
import io.github.wulkanowy.ui.base.BasePresenter
import javax.inject.Inject

class SplashPresenter @Inject constructor(private val sessionRepository: SessionRepository,
                                          errorHandler: ErrorHandler)
    : BasePresenter<SplashView>(errorHandler) {

    override fun onAttachView(view: SplashView) {
        super.onAttachView(view)
        view.run { if (sessionRepository.isSessionSaved) openMainView() else openLoginView() }
    }
}
